File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.*;

class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long start = 1, end = Long.MIN_VALUE;
        for (int time : times) {
            end = Math.max(time, end);
        }
        
        end = end * n;
        
        while (start <= end) {
            long mid = (start + end) / 2;
            long pass = 0;
            
            for (int time : times) {
                pass += (mid / time);
                if (pass >= n) break;
            }
            
            if (pass >= n) {
                answer = mid;
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        
        return answer;
        
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let end = Math.max(...times) * n;
    let start = 1;
    
    while (start <= end) {
        let mid = Math.floor((start + end) / 2);
        let count = 0;
        for (time of times) {
            count += Math.floor(mid / time);
        }
        
        if (count >= n) {
            end = mid - 1;
        } else {
            start = mid + 1;
        }
    }

    return end + 1;
}